### ISSUE TITLE
There is no error after pasting one HNS blob's URL to "Blob container or directory URL" box in Connect dialog

### ISSUE THREAD

**Issue by v-xianya** [microsoft/AzureStorageExplorer#4592]
**Storage Explorer Version**: 1.20.0-dev
**Build Number**: 202100618.1
**Branch**: rel/1.20.0
**Platform/OS**: Windows 10/ Linux Ubuntu 16.04/ MacOS Big Sur 11.3
**Architecture**: ia32/x64
**How Found**: Exploratory testing
**Regression From**: Not a regression

## Steps to Reproduce ##
1. Expand one ADLS Gen2 storage account -> Blob Containers.
2. Select a blob container which has been added a rbac role (Storage Blob Data Owner) and assigned to a azure account.
3. Upload one blob to the blob container.
4. Copy the blob's URL -> Open the connect dialog.
5. Select 'ADLS Gen2 container or directory -> Sign in using Azure Directory(Azure AD)' -> Click 'Next'.
6. Select the azure account -> Click 'Next'.
7. Paste the copied URL to 'Blob container or directory URL' box.
8. Check there is an error.

## Expected Experience ##
There is an error.

For a blobs under non-ADLS Blob Container, there is an error.

![image](https://user-images.githubusercontent.com/41351993/122517040-1abdd700-d042-11eb-94f2-574da11115d4.png)

## Actual Experience ##
There is no error.
![image](https://user-images.githubusercontent.com/41351993/122517143-40e37700-d042-11eb-8e44-0bf0a4dbea92.png)

## Additional Context ##
For the attached blob, the address bar shows below:
![image](https://user-images.githubusercontent.com/41351993/122517236-5ce71880-d042-11eb-98fb-96cba8501574.png)

**Comment by craxal** [comment]
There’s no way to determine whether the URL path refers to a blob or to a folder from the URL alone. This issue is by design.

There are some potential improvements we could consider. We could send a test request to find out for sure what kind of object it is. But that would require certain permissions depending on the API we decide to use, so it may not work in every case. It also requires a small delay for validation.

The next thing would be to post a warning message. I don’t think most users would use a file extension for a folder name (then again, folders starting with a period are pretty common). So if the path ends with an extension, we alert the user of a potential problem. This would not work, however, for files that don’t have an extension (a lot more common than folders with an extension).

In summary, there’s no one right fix for this situation. Either way, at least for now, unless we have a lot of users encountering problems with this, I think by design/won’t fix is a perfectly acceptable course of action.
